ipa-core/src/ff/prime_field.rs

```diff
@@ -53,6 +53,32 @@ pub trait PrimeField: Field + U128Conversions {
     }
 }
 
+/// Performs multiple field inversions at once for a lower cost
+/// Follows the "multiple inverses" algorithm in
+/// <https://en.wikipedia.org/wiki/Modular_multiplicative_inverse>
+/// ## Panics
+/// If any element is 0
+#[allow(non_snake_case)]
+pub fn batch_invert<P>(field_elements: &mut [P])
```

Comment:
would it be worth making it `batch_invert<const N: usize, P>(field_elements: &mut [P; N])`? that saves an allocation and works for Lagrange case

Reply:
```rust
let mut prefix_products = field_elements.to_vec();
```

`.to_vec()` needs an allocation I think. It is small and nothing crazy but if we don't need it, we could avoid it

Reply:
Oh, I actually need that allocation. The algorithm requires having a second array, but you compute that from the first one

Reply:
if it is an array in, can you put another array on the stack to hold the intermediates?

Reply:
Making sure I understand correctly: You're saying that if the function takes `[P; N]`, then you know the size at compile time and so you can avoid allocating `prefix_products` on the heap, and put it on the stack instead?

Reply:
that's right - `N` is assumed to be small enough for array to be kept on the stack.